app/serializers/vet_payment_history_serializer.rb

```diff
@@ -10,11 +10,63 @@ def id
     nil
   end
 
+  def initialize(object, options = nil)
+    @formatted_payments = []
+    @returned_payments = []
+    process_payments(object[:payments][:payment]) if object[:payments].present?
```

Comment:
This is confusing but not bc anything you did or didn't do. We may have already worked through this scenario.

IIRC object _could_ be this right? `{ payments: [], return_payments: [] }`

If this gets returned `{ payments: [], return_payments: [] }`, will the conditional work as intended?

Reply:
Yes.  If `:payments = []`, `object[:payments].present?` will return `false`.